sql/core/src/main/scala/org/apache/spark/sql/execution/ShuffledRowRDD.scala

```diff
@@ -117,7 +113,7 @@ class CoalescedPartitioner(val parent: Partitioner, val partitionStartIndices: A
 class ShuffledRowRDD(
     var dependency: ShuffleDependency[Int, InternalRow, InternalRow],
     specifiedPartitionStartIndices: Option[Array[Int]] = None)
-  extends RDD[InternalRow](dependency.rdd.context, Nil) {
+  extends RDD[InternalRow](dependency.rdd.context, Seq(dependency)) {
```

Comment:
Agree, though it only has one other use at all:

```
  private[spark] def markCheckpointed(): Unit = {
    clearDependencies()
    partitions_ = null
    deps = null    // Forget the constructor argument for dependencies too
  }
```

Right now `ShuffledRowRDD` would always 'remember' its constructor parameter separately, when this comment suggests that's not desirable. This change would get rid of that. No idea if it makes any difference -- probably doesn't either way.